### PR TITLE
Index repair

### PR DESCRIPTION
**Pull request by gguttikonda**
-----
PR for https://github.com/JanusGraph/janusgraph/issues/1895

Thank you for contributing to JanusGraph!

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there an issue associated with this PR? Is it referenced in the commit message?
- [ ] Does your PR body contain #xyz where xyz is the issue number you are trying to resolve?
- [x] Has your PR been rebased against the latest commit within the target branch (typically `master`)?
- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [x] Have you written and/or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](https://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE.txt file, including the main LICENSE.txt file in the root of this repository?
- [ ] If applicable, have you updated the NOTICE.txt file, including the main NOTICE.txt file found in the root of this repository?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?
- [ ] If this PR is a documentation-only change, have you added a `[doc only]`
  tag to the first line of your commit message to avoid spending CPU cycles in
  Travis CI when no code, tests, or build configuration are modified?

### Note:
Please ensure that once the PR is submitted, you check Travis CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by gguttikonda** [comment]
This is PR for https://github.com/JanusGraph/janusgraph/issues/1895

**Comment by FlorianHockmann** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           

Complexity increasing per file
==============================
- janusgraph-core/src/main/java/org/janusgraph/graphdb/olap/job/IndexRepairJob.java  1
- janusgraph-core/src/test/java/org/janusgraph/graphdb/database/IndexSerializerTest.java  1
         

Clones added
============
- janusgraph-core/src/test/java/org/janusgraph/graphdb/database/IndexSerializerTest.java  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/janusgraph/janusgraph/pullRequest?prid=4609964&bid=15465456)

File: janusgraph-cql/src/main/java/org/janusgraph/diskstorage/cql/CQLKeyColumnValueStore.java [hunk 1]
```diff
@@ -100,6 +103,7 @@
  * An implementation of {@link KeyColumnValueStore} which stores the data in a CQL connected backend.
  */
 public class CQLKeyColumnValueStore implements KeyColumnValueStore {
+    private static final Logger log = LoggerFactory.getLogger(CQLKeyColumnValueStore.class);
```
Comment (FlorianHockmann) [inline janusgraph-cql/src/main/java/org/janusgraph/diskstorage/cql/CQLKeyColumnValueStore.java]: Codacy Issue found: [Avoid unused private fields such as 'log'.](https://app.codacy.com/app/JanusGraph/janusgraph/pullRequest?prid=4609964)

**Comment by FlorianHockmann** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Complexity increasing per file
==============================
- janusgraph-core/src/main/java/org/janusgraph/graphdb/olap/job/IndexRepairJob.java  1
- janusgraph-core/src/test/java/org/janusgraph/graphdb/database/IndexSerializerTest.java  1
         

Clones added
============
- janusgraph-core/src/test/java/org/janusgraph/graphdb/database/IndexSerializerTest.java  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/janusgraph/janusgraph/pullRequest?prid=4609964&bid=15465456)

**Comment by porunov** [comment]
@gguttikonda Would you be able to sign EasyCLA and fix failed tests in travis?

https://travis-ci.org/JanusGraph/janusgraph/builds/620314346?utm_source=github_status&utm_medium=notification

**Comment by gani8780** [comment]
> @gguttikonda Would you be able to sign EasyCLA and fix failed tests in travis?
> 
> https://travis-ci.org/JanusGraph/janusgraph/builds/620314346?utm_source=github_status&utm_medium=notification


Thanks @porunov. I have accepted EastCLA. I'll work on travis build failures and keep you posted.

**Comment by gguttikonda** [comment]
Hi @porunov , Test failures are now fixed. Thank you for advising me on this. It helped me to change the approach a little. Below is a summary of change.

* Reverted isExhausted method in KeyIterator
* Removed the retryCount check in StandardScannerExecutor.java
    * Old
        `while (!pullThreads[i].isFinished() && retryCount < TIMEOUT_MS / TIME_PER_TRY && qr == null) { ...`     
    * New
`while (!pullThreads[i].isFinished() && qr == null) { ...`
    * Moved the `finished = true` code in DataPuller into finally block. Making sure for failure conditions or exceptions are setting the thread as finished.

File: janusgraph-cassandra/src/main/java/org/janusgraph/diskstorage/cassandra/astyanax/AstyanaxKeyColumnValueStore.java [hunk 2]
```diff
@@ -432,6 +432,7 @@ public void remove() {
             };
         }
 
+
```
Comment (farodin91) [inline janusgraph-cassandra/src/main/java/org/janusgraph/diskstorage/cassandra/astyanax/AstyanaxKeyColumnValueStore.java]: Could you remove all independent file changes?

Comment (farodin91) [reply janusgraph-cassandra/src/main/java/org/janusgraph/diskstorage/cassandra/astyanax/AstyanaxKeyColumnValueStore.java]: From the pr

File: janusgraph-cql/src/test/java/org/janusgraph/diskstorage/cql/CQLResultSetKeyIteratorTest.java [hunk 3]
```diff
@@ -16,6 +16,7 @@
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
```
Comment (FlorianHockmann) [inline janusgraph-cql/src/test/java/org/janusgraph/diskstorage/cql/CQLResultSetKeyIteratorTest.java]: Codacy Issue found: [Avoid unused imports such as 'org.junit.jupiter.api.Assertions.assertTrue'](https://app.codacy.com/manual/JanusGraph/janusgraph/pullRequest?prid=4609964)

**Comment by FlorianHockmann** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           

Complexity increasing per file
==============================
- janusgraph-core/src/test/java/org/janusgraph/graphdb/database/IndexSerializerTest.java  1
- janusgraph-core/src/main/java/org/janusgraph/graphdb/olap/job/IndexRepairJob.java  1
         

Complexity decreasing per file
==============================
+ janusgraph-core/src/main/java/org/janusgraph/diskstorage/keycolumnvalue/scan/StandardScannerExecutor.java  -1
         

Clones added
============
- janusgraph-core/src/test/java/org/janusgraph/graphdb/database/IndexSerializerTest.java  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/JanusGraph/janusgraph/pullRequest?prid=4609964&bid=15465456)

File: janusgraph-cassandra/src/main/java/org/janusgraph/diskstorage/cassandra/astyanax/AstyanaxKeyColumnValueStore.java (hunk 2 above)
Comment (gguttikonda) [reply janusgraph-cassandra/src/main/java/org/janusgraph/diskstorage/cassandra/astyanax/AstyanaxKeyColumnValueStore.java]: Hi @farodin91 , Thank you for the feedback. I checked in my changes with your review feedback.

File: janusgraph-berkeleyje/src/test/resources/log4j.properties [hunk 4]
```diff
@@ -16,12 +16,12 @@ log4j.appender.A1=org.apache.log4j.FileAppender
 log4j.appender.A1.File=target/test.log
 log4j.appender.A1.Threshold=ALL
 log4j.appender.A1.layout=org.apache.log4j.PatternLayout
-log4j.appender.A1.layout.ConversionPattern=%-4r [%t] %-5p %c %x - %m%n
```
Comment (farodin91) [inline janusgraph-berkeleyje/src/test/resources/log4j.properties]: Why did you change the log pattern of Berkeleyje?

File: janusgraph-cassandra/src/main/java/org/janusgraph/diskstorage/cassandra/embedded/CassandraEmbeddedKeyColumnValueStore.java [hunk 5]
```diff
@@ -335,6 +335,7 @@ public RowIterator(Token minimum, Token maximum, SliceQuery sliceQuery, int page
             this.nowMillis = times.getTime().toEpochMilli();
             this.keys = getRowsIterator(getKeySlice(minimum, maximum, sliceQuery, pageSize, nowMillis));
         }
+
```
Comment (farodin91) [inline janusgraph-cassandra/src/main/java/org/janusgraph/diskstorage/cassandra/embedded/CassandraEmbeddedKeyColumnValueStore.java]: nit: extra line

File: janusgraph-core/src/test/resources/log4j.properties [hunk 6]
```diff
@@ -0,0 +1,32 @@
+# Copyright 2019 JanusGraph Authors
```
Comment (farodin91) [inline janusgraph-core/src/test/resources/log4j.properties]: Could you delete this file?

File: janusgraph-cql/src/main/java/org/janusgraph/diskstorage/cql/CQLKeyColumnValueStore.java [hunk 7]
```diff
@@ -426,8 +426,9 @@ public KeyIterator getKeys(final SliceQuery query, final StoreTransaction txh) t
         private int index;
         private int paginatedResultSize;
         private final Supplier<Statement> statementSupplier;
+        private AtomicLong recordCount = new AtomicLong();
```
Comment (farodin91) [inline janusgraph-cql/src/main/java/org/janusgraph/diskstorage/cql/CQLKeyColumnValueStore.java]: Why did you done these changes?

File: janusgraph-berkeleyje/src/test/resources/log4j.properties (hunk 4 above)
Comment (gguttikonda) [reply janusgraph-berkeleyje/src/test/resources/log4j.properties]: I apologize. I was trying to isolate some test failures locally and was trying to find the timestamps for each test. I reverted this change now.

File: janusgraph-cassandra/src/main/java/org/janusgraph/diskstorage/cassandra/embedded/CassandraEmbeddedKeyColumnValueStore.java (hunk 5 above)
Comment (gguttikonda) [reply janusgraph-cassandra/src/main/java/org/janusgraph/diskstorage/cassandra/embedded/CassandraEmbeddedKeyColumnValueStore.java]: Removed :)

File: janusgraph-core/src/test/resources/log4j.properties (hunk 6 above)
Comment (gguttikonda) [reply janusgraph-core/src/test/resources/log4j.properties]: Sorry about this one! I deleted this file.

File: janusgraph-cql/src/main/java/org/janusgraph/diskstorage/cql/CQLKeyColumnValueStore.java (hunk 7 above)
Comment (gguttikonda) [reply janusgraph-cql/src/main/java/org/janusgraph/diskstorage/cql/CQLKeyColumnValueStore.java]: I was trying to keep track of how many CQL records being read from DB and trying to keep a custom metric for the same. I  forgot removing counter after I removed that part the code.

**Comment by FlorianHockmann** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           

Complexity increasing per file
==============================
- janusgraph-core/src/main/java/org/janusgraph/graphdb/olap/job/IndexRepairJob.java  1
- janusgraph-test/src/test/java/org/janusgraph/graphdb/database/IndexSerializerTest.java  1
         

Complexity decreasing per file
==============================
+ janusgraph-core/src/main/java/org/janusgraph/diskstorage/keycolumnvalue/scan/StandardScannerExecutor.java  -1
         

Clones added
============
- janusgraph-test/src/test/java/org/janusgraph/graphdb/database/IndexSerializerTest.java  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/JanusGraph/janusgraph/pullRequest?prid=4609964&bid=15465456)

File: janusgraph-cql/src/main/java/org/janusgraph/diskstorage/cql/CQLKeyColumnValueStore.java [hunk 8]
```diff
@@ -51,10 +51,10 @@
 import java.util.List;
 import java.util.Map;
 import java.util.concurrent.ExecutorService;
+import java.util.concurrent.atomic.AtomicLong;
```
Comment (FlorianHockmann) [inline janusgraph-cql/src/main/java/org/janusgraph/diskstorage/cql/CQLKeyColumnValueStore.java]: Codacy Issue found: [Avoid unused imports such as 'java.util.concurrent.atomic.AtomicLong'](https://app.codacy.com/manual/JanusGraph/janusgraph/pullRequest?prid=4609964)

**Comment by FlorianHockmann** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Complexity increasing per file
==============================
- janusgraph-core/src/main/java/org/janusgraph/graphdb/olap/job/IndexRepairJob.java  1
- janusgraph-test/src/test/java/org/janusgraph/graphdb/database/IndexSerializerTest.java  1
         

Complexity decreasing per file
==============================
+ janusgraph-core/src/main/java/org/janusgraph/diskstorage/keycolumnvalue/scan/StandardScannerExecutor.java  -1
         

Clones added
============
- janusgraph-test/src/test/java/org/janusgraph/graphdb/database/IndexSerializerTest.java  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/JanusGraph/janusgraph/pullRequest?prid=4609964&bid=15465456)

File: janusgraph-cql/src/main/java/org/janusgraph/diskstorage/cql/CQLKeyColumnValueStore.java [hunk 9]
```diff
@@ -448,15 +447,17 @@ public Row next() {
                 this.index = 0;
             }
             this.index++;
-            lastPagingState = currentResultSet.getExecutionInfo().getPagingState();
+            // Using getPagingStateUnsafe rather than getPagingState, to avoid the md5
+            // compute of the pageState, which can be quite expensive.
+            lastPagingState = currentResultSet.getExecutionInfo().getPagingStateUnsafe();
```
Comment (FlorianHockmann) [inline janusgraph-cql/src/main/java/org/janusgraph/diskstorage/cql/CQLKeyColumnValueStore.java]: How is this related to the index improvements? If it's completely unrelated, then please create a separate PR (and probably also its own issue) for this.

File: janusgraph-core/src/main/java/org/janusgraph/graphdb/olap/job/IndexUpdateJob.java [hunk 10]
```diff
@@ -139,6 +140,10 @@ public void workerIterationEnd(ScanMetrics metrics) {
             log.error("Transaction commit threw runtime exception:", e);
             metrics.incrementCustom(FAILED_TX);
             throw e;
+        } finally {
+            log.info("Index {} metrics success-tx {} doc-updates  {} success {}", indexName,
```
Comment (FlorianHockmann) [inline janusgraph-core/src/main/java/org/janusgraph/graphdb/olap/job/IndexUpdateJob.java]: (nitpick) I would change the format slightly, e.g., like this:
```
"Index {} metrics: success-tx: {} doc-updates: {} succeeded: {}"
```

File: janusgraph-core/src/main/java/org/janusgraph/graphdb/database/IndexSerializer.java [hunk 11]
```diff
@@ -355,8 +356,10 @@ public void reindexElement(JanusGraphElement element, MixedIndexType index, Map<
                 element.values(key.name()).forEachRemaining(value->entries.add(new IndexEntry(key2Field(field), value)));
             }
         }
-        final Map<String, List<IndexEntry>> documents = documentsPerStore.computeIfAbsent(index.getStoreName(), k -> Maps.newHashMap());
```
Comment (FlorianHockmann) [inline janusgraph-core/src/main/java/org/janusgraph/graphdb/database/IndexSerializer.java]: I'm not really familiar with this part of the code base, but could you quickly explain why this isn't necessary any more?

**Comment by FlorianHockmann** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           

Complexity increasing per file
==============================
- janusgraph-core/src/main/java/org/janusgraph/graphdb/olap/job/IndexRepairJob.java  1
- janusgraph-test/src/test/java/org/janusgraph/graphdb/database/IndexSerializerTest.java  2
         

Complexity decreasing per file
==============================
+ janusgraph-core/src/main/java/org/janusgraph/diskstorage/keycolumnvalue/scan/StandardScannerExecutor.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/JanusGraph/janusgraph/pullRequest?prid=4609964&bid=15465456)

File: janusgraph-test/src/test/java/org/janusgraph/graphdb/database/IndexSerializerTest.java [hunk 12]
```diff
@@ -0,0 +1,143 @@
...
+        String value = "bar";
+
+        JanusGraphElement indexableElement = mockIndexableElement(key, value, indexable);
+        ElementCategory ec = ElementCategory.VERTEX;
+
+        doReturn(ec).when(mit).getElement();
+
+        doReturn(false).when(mit).hasSchemaTypeConstraint();
+
+        Map<String, Map<String, List<IndexEntry>>> docStore = new HashMap<>();
```
Comment (FlorianHockmann) [inline janusgraph-test/src/test/java/org/janusgraph/graphdb/database/IndexSerializerTest.java]: Codacy Issue found: [Avoid unused local variables such as 'docStore'.](https://app.codacy.com/manual/JanusGraph/janusgraph/pullRequest?prid=4609964)

**Comment by FlorianHockmann** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Complexity increasing per file
==============================
- janusgraph-test/src/test/java/org/janusgraph/graphdb/database/IndexSerializerTest.java  2
- janusgraph-core/src/main/java/org/janusgraph/graphdb/olap/job/IndexRepairJob.java  1
         

Complexity decreasing per file
==============================
+ janusgraph-core/src/main/java/org/janusgraph/diskstorage/keycolumnvalue/scan/StandardScannerExecutor.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/JanusGraph/janusgraph/pullRequest?prid=4609964&bid=15465456)

File: janusgraph-core/src/main/java/org/janusgraph/graphdb/database/IndexSerializer.java (hunk 11 above)
Comment (gguttikonda) [reply janusgraph-core/src/main/java/org/janusgraph/graphdb/database/IndexSerializer.java]: Hi @FlorianHockmann,
This code is removed for 2 reasons.
1.  `getDocuments(documentsPerStore, index)` does exact same thing, and hence it was duplicate execution of `computeIfAbsent`
2. `documents` variable was not being used.

File: janusgraph-core/src/main/java/org/janusgraph/graphdb/olap/job/IndexUpdateJob.java (hunk 10 above)
Comment (gguttikonda) [reply janusgraph-core/src/main/java/org/janusgraph/graphdb/olap/job/IndexUpdateJob.java]: Updated :-)

File: janusgraph-cql/src/main/java/org/janusgraph/diskstorage/cql/CQLKeyColumnValueStore.java (hunk 9 above)
Comment (gguttikonda) [reply janusgraph-cql/src/main/java/org/janusgraph/diskstorage/cql/CQLKeyColumnValueStore.java]: It was kind of related. We are using CQL adapter, and we found that the above change is one of the biggest performance improvements while indexing. Before the above change, it was taking ~4-5 hours for ~1M vertexes and after the change, it's about 15mins. The largest time being spent was on `getPagingState` which does md5sum compute of the entire resultset.

File: janusgraph-test/src/test/java/org/janusgraph/graphdb/database/IndexSerializerTest.java [hunk 13]
```diff
@@ -0,0 +1,152 @@
...
+        String key = "foo";
+        String value = "bar";
+        Configuration config = mock(Configuration.class);
+        Serializer serializer = mock(Serializer.class);
+        Map<String, ? extends IndexInformation> indexes = new HashMap<>();
+        IndexSerializer mockSerializer = spy(new IndexSerializer(config, serializer, indexes, true));
+        StandardJanusGraphTx tx = mock(StandardJanusGraphTx.class);
+
+        JanusGraphElement indexableElement = spy(new StandardVertex(tx, 1l, ElementLifeCycle.New));
+        Property pk2 = indexableElement.property(key, value);
```
[the remark this answers is not in the document]
Comment (gguttikonda) [reply janusgraph-test/src/test/java/org/janusgraph/graphdb/database/IndexSerializerTest.java]: Refactored the code  :) and moved the mocked code into independent methods.

File: janusgraph-core/src/main/java/org/janusgraph/graphdb/database/IndexSerializer.java (hunk 11 above)
Comment (FlorianHockmann) [reply janusgraph-core/src/main/java/org/janusgraph/graphdb/database/IndexSerializer.java]: Thanks for the explanation! That makes sense of course.

File: janusgraph-test/src/test/java/org/janusgraph/graphdb/database/IndexSerializerTest.java (hunk 13 above)
[the remark this answers is not in the document]
Comment (FlorianHockmann) [reply janusgraph-test/src/test/java/org/janusgraph/graphdb/database/IndexSerializerTest.java]: Thanks, that really improved readability a lot!
(One thing I'm noticing in your recent change are some empty lines at the end of some methods which shouldn't be there, but that's not that important.)

File: janusgraph-cql/src/main/java/org/janusgraph/diskstorage/cql/CQLKeyColumnValueStore.java (hunk 9 above)
Comment (FlorianHockmann) [reply janusgraph-cql/src/main/java/org/janusgraph/diskstorage/cql/CQLKeyColumnValueStore.java]: That is of course a significant improvement, but I would still prefer to handle that outside of this PR as it's a CQL specific improvement and such a significant improvement shouldn't be "hidden" in this PR.

Comment (gguttikonda) [reply janusgraph-cql/src/main/java/org/janusgraph/diskstorage/cql/CQLKeyColumnValueStore.java]: Sure @FlorianHockmann. I reverted the change for CQLKeyColumnValueStore, and will create a new PR only for CQL optimization.

**Comment by FlorianHockmann** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Complexity increasing per file
==============================
- janusgraph-test/src/test/java/org/janusgraph/graphdb/database/IndexSerializerTest.java  2
- janusgraph-core/src/main/java/org/janusgraph/graphdb/olap/job/IndexRepairJob.java  1
         

Complexity decreasing per file
==============================
+ janusgraph-core/src/main/java/org/janusgraph/diskstorage/keycolumnvalue/scan/StandardScannerExecutor.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/JanusGraph/janusgraph/pullRequest?prid=4609964&bid=15465456)

File: janusgraph-core/src/main/java/org/janusgraph/diskstorage/keycolumnvalue/scan/StandardScannerExecutor.java [hunk 14]
```diff
@@ -146,9 +149,7 @@ public void run() {
                     SliceResult qr = queue.poll(TIME_PER_TRY,TimeUnit.MILLISECONDS); //Try very short time to see if we are done
                     if (qr==null) {
                         if (pullThreads[i].isFinished()) continue; //No more data to be expected
-                        int retryCount = 0;
-                        while (!pullThreads[i].isFinished() && retryCount < TIMEOUT_MS / TIME_PER_TRY && qr == null) {
-                            retryCount ++;
+                        while (!pullThreads[i].isFinished() && qr == null) {
```
Comment (FlorianHockmann) [inline janusgraph-core/src/main/java/org/janusgraph/diskstorage/keycolumnvalue/scan/StandardScannerExecutor.java]: Didn't the `retryCount` ensure that this doesn't result in an infinite loop? You said in the issue that this fails sometimes for vertices with very many edges, but doesn't that mean now that the `TIMEOUT_MS` are ignored?

Comment (gguttikonda) [reply janusgraph-core/src/main/java/org/janusgraph/diskstorage/keycolumnvalue/scan/StandardScannerExecutor.java]: Hi @FlorianHockmann, in my case, code was reaching to max retryCount before indexing is complete. I tried increasing the TIMEOUT_MS, but it failed too for vertexes with large number of edges. Eg: a vertex having ~500K edges, CQL KeyIterator was taking a while to return the next element, and this loop was bailing before then. Because we moved the `finished=true` in DataPuller into finally block, this should never run into an infinite loop, at the same time giving enough time for while loop to continue looking for SliceResult until DataPuller says it's finished.

**Comment by gguttikonda** [comment]
@farodin91 , @twilmes : can you please review this when you get a chance.

File: janusgraph-core/src/main/java/org/janusgraph/diskstorage/keycolumnvalue/scan/StandardScannerExecutor.java (hunk 14 above)
Comment (FlorianHockmann) [reply janusgraph-core/src/main/java/org/janusgraph/diskstorage/keycolumnvalue/scan/StandardScannerExecutor.java]: I tried to look into why this retry count was introduced and found the PR #1237 which fixed #1157.
@porunov Since you created #1157, could you take a look at this and check whether removing the timeout here could be problematic? Do you maybe have an easy way to ensure that it doesn't have negative effects on the issue you described in #1157?
Otherwise I'm OK with the explanation @gguttikonda gave and would accept this PR.

Comment (farodin91) [reply janusgraph-core/src/main/java/org/janusgraph/diskstorage/keycolumnvalue/scan/StandardScannerExecutor.java]: @porunov Do you like to comment on this?

**Comment by FlorianHockmann** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Complexity increasing per file
==============================
- janusgraph-core/src/main/java/org/janusgraph/graphdb/olap/job/IndexRepairJob.java  1
- janusgraph-test/src/test/java/org/janusgraph/graphdb/database/IndexSerializerTest.java  2
         

Complexity decreasing per file
==============================
+ janusgraph-core/src/main/java/org/janusgraph/diskstorage/keycolumnvalue/scan/StandardScannerExecutor.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/JanusGraph/janusgraph/pullRequest?prid=4609964&bid=15465456)

**Comment by FlorianHockmann** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Complexity increasing per file
==============================
- janusgraph-test/src/test/java/org/janusgraph/graphdb/database/IndexSerializerTest.java  2
- janusgraph-core/src/main/java/org/janusgraph/graphdb/olap/job/IndexRepairJob.java  1
         

Complexity decreasing per file
==============================
+ janusgraph-core/src/main/java/org/janusgraph/diskstorage/keycolumnvalue/scan/StandardScannerExecutor.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/JanusGraph/janusgraph/pullRequest?prid=4609964&bid=15465456)

**Comment by gguttikonda** [comment]
> LGTM! Thank you @gguttikonda.
> 
> Sorry, i forgot the PR fully.

Thank you @farodin91. I tried squashing all my commits into one and merged from master too. Please let me know if you have any feedback. If this PR is too lengthy, I am happy to create a new PR with all changes in one go.

**Comment by farodin91** [comment]
@gguttikonda I'm okay with squash in this PR or a different PR. It is your choice.

**Comment by gguttikonda** [comment]
> @gguttikonda I'm okay with squash in this PR or a different PR. It is your choice.
Thanks @farodin91. I created a new PR ( https://github.com/JanusGraph/janusgraph/pull/2066 ) with single commit.

File: janusgraph-core/src/main/java/org/janusgraph/diskstorage/keycolumnvalue/scan/StandardScannerExecutor.java (hunk 14 above)
Comment (porunov) [reply janusgraph-core/src/main/java/org/janusgraph/diskstorage/keycolumnvalue/scan/StandardScannerExecutor.java:152]: Sorry I didn't get back on this earlier. I will review #2066 closer to weekends because this PR is obsolete now

Comment (porunov) [reply janusgraph-core/src/main/java/org/janusgraph/diskstorage/keycolumnvalue/scan/StandardScannerExecutor.java:152]: Replied here https://github.com/JanusGraph/janusgraph/pull/2066#pullrequestreview-391922488
This change doesn't reintroduce the issue. LGTM.
